tests/functional/step_defs/test_chart_src_with_report.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""Report and chart source submission
+
+Partners or redhat associates can publish their chart by submitting
+error-free chart in source format with the report.
+"""
+import pytest
+from pytest_bdd import (
+    given,
+    scenario,
+    then,
+    when,
+)
+
+from functional.utils.chart_certification import ChartCertificationE2ETestSingle
+
+@pytest.fixture
+def workflow_test():
```

Comment:
Would like to see these values passed in as parameters from the bdd file. This would enable us to add new tests without modifying or duplicating python files.

Reply:
Just to make sure I have the correct understanding:

Refactoring the bdd files means to merge the current bdd files together and use a table for parameter settings, for example: 

  | vendor_type  | vendor    | chart   | version  |
  | partners     | hashicorp | vault   | 0.13.0   |
  | redhat       | redhat    | vault   | 0.13.0   |

and other parameters such as with/without tarball/source, with/without report?

Reply:
Or do we want just a table for the parameters that are set in the following code:
```
    test_name = 'Test Chart Source With Report'
    test_chart = 'tests/data/vault-0.13.0.tgz'
    test_report = 'tests/data/report.yaml'
```

Essentially I am asking regarding the degree of refactoring for bdd files. I tend to propose we make the current bdd files separate but parameterize the `@given` and other constant to a table in bdd files.

Reply:
Yes, refactoring the bdd files to add a parameter settings.

with/without tarball/source  - don't think these need parameters so just different givens.

For the bdd files, it would make sense to review current content, maybe some can be combined. I am not averse to one big bdd file if is makes most sense

Also, to be honest with bdd files there are a ton of different ways to do them and I am no authority but the more re-use and flexibilty we get the better. As I mention, the goal here is that we can add tests just by changing the bdd file.  
- New chart : just add it as params to the bbd file.
- New scenario : just create a flow hopefully using existing existing given, when, then but in a different combination. Some tests might need to add one or two but mostly re-use.

Reply:
Got it, thanks for explaining!